packages/components/inputs/creatable-select-input/src/creatable-select-input.tsx

```diff
@@ -222,168 +101,213 @@ CreatableSelectInput.propTypes = {
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
    */
-  noOptionsMessage: PropTypes.func,
+  noOptionsMessage?: ReactSelectProps['noOptionsMessage'];
   /**
    * Handle blur events on the control
    */
-  onBlur: PropTypes.func,
+  onBlur?: (event: TEvent) => void;
   /**
    * Called with a fake event when value changes. The event's `target.name` will be the `name` supplied in props. The event's `target.value` will hold the value. The value will be the selected option, or an array of options in case `isMulti` is `true`.
    * <br />
    * Signature: `(event) => void`
    */
-  onChange: PropTypes.func.isRequired,
+  onChange: (event: TEvent, info: ActionMeta<unknown>) => void;
   /**
    * Handle focus events on the control
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
    */
-  onFocus: PropTypes.func,
+  onFocus?: ReactSelectProps['onFocus'];
   /**
    * Handle change events on the input
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
    */
-  onInputChange: PropTypes.func,
+  onInputChange?: ReactSelectProps['onInputChange'];
   /**
    * Array of options that populate the select menu
    */
-  options: PropTypes.arrayOf(
-    PropTypes.oneOfType([
-      PropTypes.shape({ value: PropTypes.string.isRequired }),
-      PropTypes.shape({
-        options: PropTypes.arrayOf(
-          PropTypes.shape({ value: PropTypes.string.isRequired })
-        ),
-      }),
-    ])
-  ),
+  options: TOptions;
   /**
    * Determines if option groups will be separated by a divider
    */
-  showOptionGroupDivider: PropTypes.bool,
+  showOptionGroupDivider?: boolean;
   /**
    * Placeholder text for the select value
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
    */
-  placeholder: PropTypes.string,
+  placeholder?: ReactSelectProps['placeholder'];
   /**
    * Sets the tabIndex attribute on the input
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
    */
-  tabIndex: PropTypes.string,
+  tabIndex?: ReactSelectProps['tabIndex'];
   /**
    * Select the currently focused option when the user presses tab
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
    */
-  tabSelectsValue: PropTypes.bool,
+  tabSelectsValue?: ReactSelectProps['tabSelectsValue'];
   /**
    * The value of the select; reflected by the selected option
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
    */
-  value: (props, ...rest) =>
-    props.isMulti
-      ? PropTypes.arrayOf(
-          PropTypes.shape({ value: PropTypes.string.isRequired })
-        )(props, ...rest)
-      : PropTypes.shape({ value: PropTypes.string.isRequired })(props, ...rest),
+  value: PropsValue<TValue>;
 
   // Creatable props
```

Comment:
All these props here are also from react-select right? https://react-select.com/props#creatable-props

Can you just reference them as we do with other props, e.g. `ReactSelectProps['allowCreateWhileLoading']`?

Also provide the link to the react-select docs.

Reply:
They are currently referenced from react-select, and links to the docs are provided.

Reply:
updated.